documents/coredocument.go

```diff
@@ -457,9 +504,82 @@ func (cd *CoreDocument) GetCollaborators(filterIDs ...identity.DID) ([]identity.
 	return filterCollaborators(cs, filterIDs...), nil
 }
 
+// GetCollaborators1 returns the collaborators excluding the filteredIDs
+func (cd *CoreDocument) GetCollaborators1(filterIDs ...identity.DID) (CollaboratorsAccess, error) {
+
+	rcs, err := cd.getReadCollaborators(coredocumentpb.Action_ACTION_READ_SIGN, coredocumentpb.Action_ACTION_READ)
+	if err != nil {
+		return CollaboratorsAccess{}, err
+	}
+	wcs, err := cd.getWriteCollaborators(coredocumentpb.TransitionAction_TRANSITION_ACTION_EDIT)
+	if err != nil {
+		return CollaboratorsAccess{}, err
+	}
+	rc := filterCollaborators(rcs, filterIDs...)
+	wc := filterCollaborators(wcs, filterIDs...)
+
+	collabs := CollaboratorsAccess{
+		ReadCollaborators:      rc,
+		ReadWriteCollaborators: wc,
+	}
+
+	return collabs, nil
+}
+
+// TODO: remove this method with new GetCollaborators
 // getCollaborators returns all the collaborators who belongs to the actions passed.
 func (cd *CoreDocument) getCollaborators(actions ...coredocumentpb.Action) (ids []identity.DID, err error) {
-	findRole(cd.Document, func(_, _ int, role *coredocumentpb.Role) bool {
+	findReadRole(cd.Document, func(_, _ int, role *coredocumentpb.Role) bool {
```

Comment:
this shouldn't return a bool but instead the err itself. Once you return the error, we should return the error to the caller else we will just swallow the error